forge-gui/res/cardsfolder/upcoming/become_the_pilot.txt

```diff
@@ -0,0 +1,9 @@
+Name:Become the Pilot
+ManaCost:3 U U
+Types:Enchantment Aura
+K:Enchant noncommander creature
+A:SP$ Attach | ValidTgts$ Creature.IsNotCommander | TgtPrompt$ Select target noncommander creature | AILogic$ GainControl
+S:Mode$ Continuous | Affected$ Card.EnchantedBy | GainControl$ You | Description$ You control enchanted creature.
+S:Mode$ Continuous | Affected$ Card.EnchantedBy | AddPower$ 2 | IsPresent$ Card.EnchantedBy+attacking EnchantedOwner,Card.EnchantedBy+attacking Permanent.EnchantedOwnerCtrl | PresentCompare$ EQ0 | AddToughness$ 2 | Description$ Enchanted creature gets +2/+2 and can't be blocked unless it's attacking its owner or a permanent its owner controls.
+S:Mode$ CantBlockBy | ValidAttacker$ Card.EnchantedBy | IsPresent$ Card.EnchantedBy+attacking Player.EnchantedOwner,Card.EnchantedBy+attacking Permanent.EnchantedOwnerCtrl | PresentCompare$ EQ0 | Secondary$ True | Description$ Enchanted creature gets +2/+2 and can't be blocked unless it's attacking its owner or a permanent its owner controls.
```

Comment:
EnchantedOwner does not exist (yet)
also use ValidDefender$

Reply:
Hmm I think Enslave is wrong then ?
![image](https://github.com/Card-Forge/forge/assets/67333662/cd36f136-6eee-4884-a57c-9318c1e32702)
****

Reply:
you need to differentiate between Defined and Valid properties
sometimes they share an expression and sometimes you can move from one to the other
in this case something like this might have worked:
```
+attacking EnchantedOwner
+attacking Valid Permanent.ControlledBy EnchantedOwner
```

but I think the second might not be split correctly

Reply:
Try calling split with limit=2 here:
https://github.com/Card-Forge/forge/blob/269eb6a1f0c6b3aa0548c7b69b9ad08ff49bbc81/forge-game/src/main/java/forge/game/card/CardProperty.java#L1632